sdk/inc/azure/iot/az_iot_pnp_client.h

```diff
@@ -658,10 +658,15 @@ AZ_NODISCARD az_result az_iot_pnp_client_property_builder_end_reported_status(
 /**
  * @brief Read the IoT Plug and Play property version.
  *
+ * @warning This modifies the state of the json reader. To then use the same json reader
+ * with az_iot_pnp_client_property_get_next_component_property(), you must call
+ * az_json_reader_init() again after this call and before the call to
```

Comment:
How? What if I don't have the JSON data to pass to `_init()` anymore and all I have is the reader?

Reply:
The question isn't about the overhead, it is about feasibility.  If I, as a the caller, only have a json reader (someone else gave  it to me, who also owns the data and promises to keep it live), how am I supposed to re-create the json reader by calling az_json_reader_init?

> If you don't have access to the data anymore you're kinda out of luck.

And we are OK with that behavior for the user as a trade-off? Seems like a decision to over-optimize for stack size given the potential negative impact for the caller.

Reply:
Good question

If a caller has access to the json reader, in a state that is as required, then they themselves can make a copy via
```c
az_json_reader copy_json = *passed_json;
```
before they call the first API. At that point, they are taking the same hit on stack usage as before. They can use a separate json reader for each call at that point.

Reply:
Gotcha. In that case, let's document that explicitly (make a copy of the reader) rather than asking the caller to call `init`. Or maybe have both, if you really want to call out the init use-case where they own/have access to the data.